tests/conftest.py

```diff
@@ -34,32 +35,34 @@ class TestbedInfo(object):
 
     def __init__(self, testbed_file):
         self.testbed_filename = testbed_file
-        self.testbed_topo = {}
+        self.testbed_topo = defaultdict()
+        CSV_FIELDS = ('conf-name', 'group-name', 'topo', 'ptf_image_name', 'ptf', 'ptf_ip', 'server', 'vm_base', 'dut', 'comment')
 
         with open(self.testbed_filename) as f:
-            topo = csv.DictReader(f)
+            topo = csv.DictReader(f, fieldnames=CSV_FIELDS)
+
+            # Validate all field are in the same order and are present
+            header = next(topo)
+            for field in CSV_FIELDS:
+                assert header[field].replace('#', '').strip() == field
+
             for line in topo:
-                tb_prop = {}
-                name = ''
-                for key in line:
-                    if ('uniq-name' in key or 'conf-name' in key) and '#' in line[key]:
-                        continue
-                    elif 'uniq-name' in key or 'conf-name' in key:
-                        name = line[key]
-                    elif 'ptf_ip' in key and line[key]:
-                        ptfaddress = ipaddress.IPNetwork(line[key])
-                        tb_prop['ptf_ip'] = str(ptfaddress.ip)
-                        tb_prop['ptf_netmask'] = str(ptfaddress.netmask)
-                    elif key == 'topo':
-                        tb_prop['topo'] = {}
-                        tb_prop['topo']['name'] = line[key]
-                        with open("../ansible/vars/topo_{}.yml".format(tb_prop['topo']['name']), 'r') as fh:
-                            tb_prop['topo']['properties'] = yaml.safe_load(fh)
-                    else:
-                        tb_prop[key] = line[key]
-
-                if name:
-                    self.testbed_topo[name] = tb_prop
+                if '#' in line['conf-name']:
```

Comment:
Should this be startswith() after strip()?

Reply:
This is doable. Existing logic didn't do so as could be seen here:

https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/test_facts.py#L115

The logic was simply relocated to different place in order skip the line rather than skipping this particular key/field only.